presto-orc/src/main/java/com/facebook/presto/orc/OrcWriter.java

```diff
@@ -295,6 +293,8 @@ private void writeChunk(Page chunk)
         // convert dictionary encoded columns to direct if dictionary memory usage exceeded
         dictionaryCompressionOptimizer.optimize(bufferedBytes, stripeRowCount);
 
+        retainedBytes = toIntExact(columnWriters.stream().mapToLong(ColumnWriter::getRetainedBytes).sum());
+
         if (stripeRowCount == stripeMaxRowCount || retainedBytes > stripeMaxBytes || dictionaryCompressionOptimizer.isFull()) {
```

Comment:
It looks like you have change the code so that you are now doing `bufferedBytes > stripeMaxBytes` instead of `retainedBytes > stripeMaxBytes` despite your reply to my comment suggesting otherwise. If this is not the case, we need to sort out what actually happened.